partner_firstname/res_user.py

```diff
@@ -35,8 +35,14 @@ def create(self, cr, user, vals, context=None):
         installed
         """
         vals2 = vals.copy()
+
+        res_partner = self.pool.get('res.partner')
+        partner = res_partner.browse(cr, user, vals2['partner_id'], context)
+
         if 'name' in vals:
             vals2['lastname'] = vals2['name']
+        elif partner.lastname:
```

Comment:
The line below takes into account that 'lastname' can be present in vals. Do you need to do that too?